cdap-app-fabric/src/main/java/io/cdap/cdap/app/task/SupportBundleMetricsCollectionTask.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.app.task;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.metrics.query.MetricsQueryHelper;
+import io.cdap.cdap.proto.MetricQueryResult;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/** Collects metric from single pipeline run */
+public class SupportBundleMetricsCollectionTask {
```

Comment:
All tasks should implement a common interface and communicate with job through it